src/main/java/org/opentripplanner/routing/algorithm/strategies/InterleavedBidirectionalHeuristic.java

```diff
@@ -87,6 +170,20 @@
     // True when the entire transit network has been explored by the reverse search.
     boolean finished = false;
 
+    // The speed for calculating the pre-transit remaining weight with a euclidean heuristic
+    private double remainingDistanceSpeed;
+
+    // A cache of pre-transit remaining weights based on a vertex as a key
+    private Map<Vertex, Double> preTransitRemainingWeightEstimates;
+
+    // A threshold in meters beyond which the pre-transit euclidean heuristic is applied in order
+    // to save time with trip calculations
+    private final double PRE_TRANSIT_EUCLIDEAN_WALK_DISTANCE_THRESHOLD = 5000; // TODO determine a good value empirically
```

Comment:
This value seems to be ok but perhaps a more optimized value could be found through testing but I didn't have time to do that now and maybe someone else has a better test setup for that purpose. It is also fine for me if this pull request gets merged with the current value.